### PR TITLE
C3 nohahesham s18 share projects

### PR DESCRIPTION
**Pull request by nohahesham**
plz review code and documentation 
@Omar-Hossam 
@SalmaFG 
@m0hamed 
share and remove stories in this branch
[issue #204][issue #207]

### CODE REVIEW

**Comment by NourhanMohamed** [comment]
@nohahesham I changed the code reviewer to @Omar-Hossam because he's now your reviewer according to the new reviewing system they chose :)

**Comment by nohahesham** [comment]
okk:)

**Comment by omar-mohyeldin** [comment]
@nohahesham please specify which files you have edited, and modify your code to satisfy our ruby code conventions to be able to review

File: arability/app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -1,66 +1,556 @@
...
+      respond_to do |format|
+        format.html
+        format.json { render json: @project }
+      end
+    else
+      developer_unauthorized
+      render 'pages/home'
+    end
+  end
+  # Author:
```
Comment (salmafg) [inline arability/app/controllers/projects_controller.rb:90]: Enter a line before the documentation.

File: arability/app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -1,66 +1,556 @@
...
+      flash[:notice] = "Project has been successfully updated."
+    else
+      render :action => 'edit'
+    end
+  else
+    developer_unauthorized
+  end
+end
+
+ # author:Noha hesham
```
Comment (salmafg) [inline arability/app/controllers/projects_controller.rb:150]: Enter author name in another line.

**Comment by nohahesham** [comment]
developer controller ,share.html.erb @Omar-Hossam

File: arability/app/controllers/developer_controller.rb [hunk 3]
```diff
@@ -43,6 +43,69 @@ def create
...
+  #  Noha Hesham
+  # Description:
+  #  It removes the project which was shared with the developer 
+  #  by finding this developer and the project to be removed
+  # params:
+  #  none
+  # success:
+  #  project is unshared successfully
+  # failure:
+  # project is not removed
```
Comment (salmafg) [inline arability/app/controllers/developer_controller.rb:56]: Please indent the failure scenario.

File: arability/app/controllers/developer_controller.rb [hunk 4]
```diff
@@ -43,6 +43,69 @@ def create
...
+  # success:
+  #  project is unshared successfully
+  # failure:
+  # project is not removed  
+  def remove_developer_from_project
+    dev = Developer.find(params[:dev_id])
+    project = Project.find(params[:project_id])
+    project.developers_shared.delete(dev)
+    project.save
+    flash[:notice] = flash[:notice] = I18n.t('controller.my_subscription.flash_messages.developer_unshared')
```
Comment (omar-mohyeldin) [inline arability/app/controllers/developer_controller.rb:62]: use " instead of '

File: arability/app/controllers/developer_controller.rb [hunk 5]
```diff
@@ -43,6 +43,69 @@ def create
...
+    redirect_to :action => "share",:controller => "projects", :id => params[:project_id]
```
Comment (omar-mohyeldin) [inline arability/app/controllers/developer_controller.rb:63]: divide line to be less than 80 characters

File: arability/app/controllers/developer_controller.rb [hunk 6]
```diff
@@ -43,6 +43,69 @@ def create
...
+  # success:
+  #  project is shared successfully
+  # failure:
+  #  project is not shared with the developer ,and doesnt appear in his 
+  #  shared projects tab
+  def share_project_with_developer
+    @project = Project.find(params[:id])
+    gamer = Gamer.find_by_email(params[:email])
+    if(!gamer.present?)
+      flash[:notice] = I18n.t('controller.my_subscription.flash_messages.Email_doesnt_exist')
```
Comment (omar-mohyeldin) [inline arability/app/controllers/developer_controller.rb:81]: use " instead of '

File: arability/app/controllers/developer_controller.rb [hunk 7]
```diff
@@ -43,6 +43,69 @@ def create
...
+      else
+       developer = Developer.find_by_gamer_id(gamer.id)
+        if developer == nil
+         flash[:notice] = I18n.t('controller.my_subscription.flash_messages.Email_for_a_gamer_not_a_developer')
```
Comment (omar-mohyeldin) [inline arability/app/controllers/developer_controller.rb:85]: use " instead of '

File: arability/app/controllers/developer_controller.rb [hunk 8]
```diff
@@ -43,6 +43,69 @@ def create
...
+        else
+         developer2 = Developer.find_by_gamer_id(current_gamer.id)
+        if developer == developer2 
+          flash[:notice] = I18n.t('controller.my_subscription.flash_messages.you_cant_share_the_project_with_yourself')
```
Comment (omar-mohyeldin) [inline arability/app/controllers/developer_controller.rb:89]: use " instead of '

File: arability/app/controllers/developer_controller.rb [hunk 9]
```diff
@@ -43,6 +43,69 @@ def create
...
+          redirect_to "/developers/projects/#{@project.id}/share"
+          return
+        end
+        if(SharedProject.where("project_id = ? and developer_id = ?", @project.id, developer.id).size > 0)
```
Comment (omar-mohyeldin) [inline arability/app/controllers/developer_controller.rb:93]: divide line to be less than 80 characters

File: arability/app/controllers/developer_controller.rb [hunk 10]
```diff
@@ -43,6 +43,69 @@ def create
...
+          flash[:notice] = I18n.t('controller.my_subscription.flash_messages.already_shared')
```
Comment (omar-mohyeldin) [inline arability/app/controllers/developer_controller.rb:94]: use " instead of '

File: arability/app/controllers/developer_controller.rb [hunk 11]
```diff
@@ -43,6 +43,69 @@ def create
...
+        else
+          developer.projects_shared << @project
+           if(developer.save)
+            flash[:notice] = I18n.t('controller.my_subscription.flash_messages.project_has_been_successfully_shared')
```
Comment (omar-mohyeldin) [inline arability/app/controllers/developer_controller.rb:98]: use " instead of '

File: arability/app/controllers/developer_controller.rb [hunk 12]
```diff
@@ -43,6 +43,69 @@ def create
...
+            redirect_to :action => "share",:controller => "projects", :id => params[:id]
+            return
+           else
+            flash[:notice] = I18n.t('controller.my_subscription.flash_messages.cant_share_the_project')
```
Comment (omar-mohyeldin) [inline arability/app/controllers/developer_controller.rb:102]: use " instead of '

File: arability/app/views/projects/share.html.erb [hunk 13]
```diff
@@ -0,0 +1,34 @@
+<div class="well" style="height:500px;">
+	<%= form_for @project, :url => "/developers/projects/share_project_with_developer" do |f| %>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/share.html.erb:2]: use one: two instead of :one => two

File: arability/app/views/projects/share.html.erb [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+  <%= hidden_field_tag "id", @project.id%>
+  <%= f.label "#{t(:email2)}"%>
+  <div class="form-search">
+  <%= text_field :email,"", :placeholder=>"Please enter an email"%>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/share.html.erb:6]: use one: two instead of :one => two

File: arability/app/views/projects/share.html.erb [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+  <%= f.submit "#{t(:share)}", :class => "btn btn-success"%>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/share.html.erb:7]: use one: two instead of :one => two

**Comment by omar-mohyeldin** [comment]
@nohahesham please modify your code

**Pull request merged**